core/src/main/java/org/apache/accumulo/core/security/crypto/NonCachingSecretKeyEncryptionStrategy.java

```diff
@@ -81,7 +81,10 @@ private void doKeyEncryptionOperation(int encryptionMode, CryptoModuleParameters
       Cipher cipher = DefaultCryptoModuleUtils.getCipher(params.getAllOptions().get(Property.CRYPTO_DEFAULT_KEY_STRATEGY_CIPHER_SUITE.getKey()));
 
       try {
-        cipher.init(encryptionMode, new SecretKeySpec(keyEncryptionKey, params.getAlgorithmName()));
+    	if (keyEncryptionKey.length > 0)
```

Comment:
The original reported issue is that the return value of the read is not captured or checked. The conditional being checked here only verifies that the allocated array was larger than an empty array. What it actually needs to check is that the number of bytes read into the array is the same as the value of the length field, regardless of what else we want to check. The conditional in this patch is sufficient only if we switch to using `readFully` instead of `read`, which throws an EOFException if the number of bytes read from the input stream isn't enough to fill the buffer (e.g. doesn't match the expected keyEncryptionKeyLength).